### PR TITLE
Remove various pylint disable statements

### PR DESCRIPTION
**Pull request by troydai**
I pick the pylint disable statements with fewer occurrences and begin eliminating them. In most of the cases, the statements are not needed anymore. And for others, I made some minor change.

### CODE REVIEW

File: src/azure-cli-core/tests/test_command_registration.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def tearDownClass(cls):
         logging.shutdown()
 
     @staticmethod
-    def sample_vm_get(resource_group_name, vm_name, opt_param=None, expand=None, custom_headers={},  # pylint: disable=dangerous-default-value, too-many-arguments
-                      raw=False, **operation_config):
+    def sample_vm_get(resource_group_name, vm_name, opt_param=None, expand=None, custom_headers=None, raw=False,
+                      **operation_config):
```
Comment (troydai) [inline src/azure-cli-core/tests/test_command_registration.py]: Note: this is a very bad violation. Default parameter value shouldn't be mutable.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 # TODO Move this to a package shared by CLI and SDK
 from functools import total_ordering
+from importlib import import_module
 from enum import Enum
-from importlib import import_module  # pylint: disable=wrong-import-order
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py]: This is going to cause a wrong-import-order pylint error on whichever version of Python you aren't using...

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py [hunk 3]
```diff
@@ -18,8 +16,8 @@ def build_table_output(result, projection):
     final_list = []
 
     for item in result:
-        def _value_from_path(path):
-            obj = item  # pylint: disable=cell-var-from-loop
+        def _value_from_path(each_item, path):
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py]: @tjprescott please take another look. Assigning value to function from out of the closure is often misleading and error-prone

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 2 above)
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py]: Well, looks like the problem is solved: https://travis-ci.org/troydai/azure-cli/builds/238432295.

File: src/azure-cli-core/tests/test_command_registration.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/tests/test_command_registration.py]: How did this work before without the `pass` statement below?

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py [hunk 4]
```diff
@@ -665,8 +662,7 @@ def add_certificate_contact(client, vault_base_url, contact_email, contact_name=
 
 def delete_certificate_contact(client, vault_base_url, contact_email):
     """ Remove a certificate contact from the specified vault. """
-    from azure.keyvault.models import \
-        (Contacts, KeyVaultErrorException)
+    from azure.keyvault.models import Contacts
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: So `KeyVaultErrorException` is not used in this file anymore? just checking..

Comment (troydai) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: No. Pylint promotes the non-use error.

**Comment by troydai** [comment]
We should use `enable=useless-suppression`

File: src/azure-cli-core/tests/test_command_registration.py [hunk 5]
```diff
@@ -50,6 +50,7 @@ def sample_vm_get(resource_group_name, vm_name, opt_param=None, expand=None, cus
         :rtype: VirtualMachine
         :rtype: msrest.pipeline.ClientRawResponse if raw=True
         """
+        pass
```
Comment (troydai) [inline src/azure-cli-core/tests/test_command_registration.py]: I have no idea why this worked before.

**Comment by troydai** [comment]
@tjprescott @derekbekoe looks good?

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 2 above)
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py]: Yay for small miracles! 🌈

File: src/azure-cli-core/tests/test_command_registration.py (hunk 1 above)
Comment (tjprescott) [inline src/azure-cli-core/tests/test_command_registration.py]: I don't think anything ever changed the custom headers, so the dangerous behavior never manifested.

File: src/azure-cli-core/tests/test_profile.py [hunk 6]
```diff
@@ -852,7 +852,7 @@ def __exit__(self, _2, _3, _4):
 
 class SubscriptionStub(Subscription):  # pylint: disable=too-few-public-methods
 
-    def __init__(self, id, display_name, state, tenant_id):  # pylint: disable=redefined-builtin,
+    def __init__(self, id, display_name, state, tenant_id):  # pylint: disable=redefined-builtin
```
Comment (tjprescott) [inline src/azure-cli-core/tests/test_profile.py]: Why not fix this one?

File: src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_command_type.py [hunk 7]
```diff
@@ -728,13 +728,15 @@ def _resolve_conflict(self,
...
-    def _flatten_object(self, path, param_model,  # pylint: disable=dangerous-default-value
-                        conflict_names=[]):
+    def _flatten_object(self, path, param_model, conflict_names=None):
         """Flatten a complex parameter object into command line arguments.
         :param str path: The complex parameter namespace.
         :param class param_model: The complex parameter class.
         :param list conflict_name: List of argument names that conflict.
         """
+        if conflict_names is None:
+            conflict_names = []
```
Comment (tjprescott) [inline src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_command_type.py]: Can simplify to:
```Python
conflict_names = conflict_names or []
```

File: src/azure-cli-core/tests/test_profile.py (hunk 6 above)
Comment (troydai) [inline src/azure-cli-core/tests/test_profile.py]: Baby steps :) 
one type of rules at a time.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py (hunk 3 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_format.py]: Your change LGTM

**Comment by troydai** [comment]
Will merge after CI passes.

**Pull request merged**